media-translation/cloud-client/translate_from_file.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Google Cloud Media Translation sample application.
+Example usage:
+    python translate_from_file.py resources/audio.raw
+"""
+
+import argparse
+
+# [START media_translation_translate_from_file]
+
+
+def translate_from_file(file_path):
+    from google.cloud import mediatranslation
+
+    client = mediatranslation.SpeechTranslationServiceClient()
+
+    config = mediatranslation.StreamingTranslateSpeechConfig(
+        audio_config=mediatranslation.TranslateSpeechConfig(
+            audio_encoding='linear16',
+            source_language_code='en-US',
+            target_language_code='fr-FR'))
+
+    def request_generator(config, audio_file_path):
+
+        # The first request contains the configuration.
+        # Note that audio_content is explicitly set to None.
+        yield mediatranslation.StreamingTranslateSpeechRequest(
+            streaming_config=config, audio_content=None)
+
+        with open(audio_file_path, 'rb') as audio:
+            while True:
+                chunk = audio.read(4096)
+                if not chunk:
+                    break
+                yield mediatranslation.StreamingTranslateSpeechRequest(
+                    audio_content=chunk,
+                    streaming_config=config)
+
+    requests = request_generator(config, file_path)
+    responses = client.streaming_translate_speech(requests)
+
+    for response in responses:
+        # Once the transcription settles, the response contains the
+        # is_final result. The other results will be for subsequent portions of
+        # the audio.
```

Comment:
Do you need to set something in the config for partial transcription?

Reply:
No.